### PR TITLE
1852: Make it possible to disable backport PRs for repository

### PR DESCRIPTION
**Pull request by zhaosongzs**
In this patch, PR bot has been added with the capability to control whether backport pr is allowed in specific repositories.

By default, backport is enabled.

However, if it is desired to disable backport in a repo, the configuration can be added to the PR bot configuration as follows:
```
{
  "pr": {
      "repositories": {
          "repo1": {
               "backport": false
          }
       }
  }
}
```

If backport pr is disabled and the title of the pr matches the pattern of backport pr, the pr bot would reply a warning message to the user and prompt for the next steps.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-1852](https://bugs.openjdk.org/browse/SKARA-1852): Make it possible to disable backport PRs for repository

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara.git pull/1497/head:pull/1497` \
`$ git checkout pull/1497`

Update a local copy of the PR: \
`$ git checkout pull/1497` \
`$ git pull https://git.openjdk.org/skara.git pull/1497/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1497`

View PR using the GUI difftool: \
`$ git pr show -t 1497`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1497.diff">https://git.openjdk.org/skara/pull/1497.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/skara/pull/1497#issuecomment-1498268318)

### CODE REVIEW

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java [hunk 1]
```diff
@@ -290,14 +290,27 @@ public Collection<WorkItem> prRun(Path scratchPath) {
         var activeReviews = CheckablePullRequest.filterActiveReviews(allReviews, pr.targetRef());
         // Determine if the current state of the PR has already been checked
         if (forceUpdate || !currentCheckValid(census, comments, activeReviews, labels)) {
+            var backportHashMatcher = BACKPORT_HASH_TITLE_PATTERN.matcher(pr.title());
+            var backportIssueMatcher = BACKPORT_ISSUE_TITLE_PATTERN.matcher(pr.title());
+
+            var backportDisabledText = "<!-- backport error -->\n" +
+                    ":warning: @" + pr.author().username() + " backport PR is not allowed in this repository, please close this pr." +
+                    " If it was unintentional, please modify the title of this PR.";
```
Comment (erikj79) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java:298]:
```suggestion
                    ":warning: @" + pr.author().username() + " backports are not allowed in this repository." +
                    " If it was unintentional, please modify the title of this pull request.";
```

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java [hunk 2]
```diff
@@ -290,14 +290,27 @@ public Collection<WorkItem> prRun(Path scratchPath) {
...
+
+            if (!bot.enableBackport() && (backportHashMatcher.matches() || backportIssueMatcher.matches())) {
+                addErrorComment(backportDisabledText, comments);
+                return List.of();
+            }
+
+            removeErrorComment(backportDisabledText, comments);
```
Comment (erikj79) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java:305]: No need to remove the comment.

Comment (zhaosongzs) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java:305]: Will fix it

**Review by erikj79 — APPROVED** [review]

**Pull request closed without merging**